addons/website_event/static/src/js/website_event.editor.js

```diff
@@ -2,11 +2,201 @@ odoo.define('website_event.editor', function (require) {
+                    var that = this;
+                    // fetch data only once and store it
+                    if (!this.selection_data) {
+                        self._rpc({
+                            model: 'res.partner',
+                            method: 'search_read',
+                            domain: [['company_id', '=', false]],
+                            fields: ['display_name', 'street', 'zip', 'city', 'country_id'],
+                        }).then(function (data) {
+                            that.fill_data(query, data);
+                            that.selection_data = data;
+                        });
+                    } else {
+                        this.fill_data(query, this.selection_data);
+                    }
+                }
+            });
+        });
+    },
+    _initDateTimePicker: function ($dateGroup) {
+        var datetimepickerFormat = time.getLangDatetimeFormat();
+        var minDate = moment().subtract(1, "d");
+        var maxDate = moment().add(200, "y");
+        var disabledDates = [minDate, maxDate];
+
+        $dateGroup.datetimepicker({
+            format : datetimepickerFormat,
+            minDate: minDate,
+            maxDate: maxDate,
+            disabledDates: disabledDates,
+            useCurrent: false,
+            viewDate: moment(new Date()).hours(minDate.hours()).minutes(minDate.minutes()).seconds(minDate.seconds()).milliseconds(minDate.milliseconds()),
+            calendarWeeks: false,
+            icons: {
+                time: 'fa fa-clock-o',
+                date: 'fa fa-calendar',
+                next: 'fa fa-chevron-right',
+                previous: 'fa fa-chevron-left',
+                up: 'fa fa-chevron-up',
+                down: 'fa fa-chevron-down',
+            },
+            locale : moment.locale(),
+            allowInputToggle: true,
+        });
+        $dateGroup.on('error.datetimepicker', function (err) {
+            if (err.date) {
+                if (err.date < minDate) {
+                    Dialog.alert(this, _t('The date you selected is lower than the minimum date: ') + minDate.format(datetimepickerFormat));
+                }
+
+                if (err.date > maxDate) {
+                    Dialog.alert(this, _t('The date you selected is greater than the maximum date: ') + maxDate.format(datetimepickerFormat));
+                }
+            }
+            return false;
+        });
+    },
+    _validateDates: function (values, errors) {
+        if (!values.eventStart) {
+            errors.push('event_start');
+        } else if (!values.eventEnd) {
+            errors.push('invalid_end');
+        } else {
+            var momentStartDate = moment(values.eventStart);
+            var momentEndDate = moment(values.eventEnd);
+            if (!momentEndDate.isAfter(momentStartDate)) {
+                errors.push('invalid_end');
+            }
+        }
+    },
+    _validateForm: function (values){
```

Comment:
Generic guideline remark: don't forget to order by public / private / handlers and then alphabetically (+ some docstrings).